datafusion/sql/src/expr/binary_op.rs

```diff
@@ -47,6 +47,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             BinaryOperator::PGBitwiseShiftRight => Ok(Operator::BitwiseShiftRight),
             BinaryOperator::PGBitwiseShiftLeft => Ok(Operator::BitwiseShiftLeft),
             BinaryOperator::StringConcat => Ok(Operator::StringConcat),
+            BinaryOperator::PGOverlap => Ok(Operator::And),
```

Comment:
Postgres array overlap (`&&`) returns a boolean, which is not same as `array_intersect`/`list_intersect` in Duckdb. But looks like you are treating `And` (`&&`) as an alias like Duckdb. Maybe you should not parse a `PGOverlap` to `And`.

Reply:
Agree, maybe we should introduce new Operator for `&&`.

Reply:
I pass the test without this, I dont think we need this 🤔

Reply:
@viirya I think PGOverlap is just a parsed token. What does this token do (return boolean or list_intersect) should be define latter on. It is fine to convert it to Operator::And.

Reply:
btw PGOverlap does not seem pg specific, although the comment said. It is also parsed in GenericDialect. Is the comment misleading or what? 🤔 

https://github.com/sqlparser-rs/sqlparser-rs/blob/8d97330d42fa9acf63d8bf6412b762948e025496/src/parser/mod.rs#L2106-L2107

Reply:
No...this PR makes `AND` as an alias operator of `array_intersect` function when it is applied with `List` parameters. So this line actually means an `BinaryOperator::PGOverlap` will be treated as `AND` operator in DataFusion (i.e., `AND` in DataFusion is compatible with Postgres' ``PGOverlap` operator).

Reply:
> I think PGOverlap is just a parsed token. What does this token do (return boolean or list_intersect) should be define latter on. It is fine to convert it to Operator::And.

Doesn't this PR already define `AND` as an alias of `array_intersect`?

Reply:
What is the concern of treating pgoverlap as Operator::AND

Reply:
I agree with @jayzhan211 
>  I think it is fine that the return value is different based on the input types.

Reply:
There are questions we should consider:

1. Should we make `AND` as an alias of `array_intersect`?
2. Should we make `&&` as an alias of `AND` and so `array_intersect` too?

This PR currently seems to do both 1 and 2.

From the doc link (https://duckdb.org/docs/sql/functions/nested) in this ticket, I only see DuckDB uses `&&` as alias operator for `list_intersect`.  Is `AND` also an alias of `list_intersect` in DuckDB? I don't see it in the doc.

In Postgres, neither `&&` nor `AND` is used as alias of `array_intersect` (actually seems it doesn't have this function). `&&` is an alias of array overlapping.

I'm not sure what engine DataFusion follow in general, but it looks like this doesn't follow completely DuckDB (if `AND` is not alias), and not Postgres at all. It looks it is inconsistent.

Reply:
We should only have `list && list` for array intersect not `AND` to be consistent with Duckdb.

I see, BinaryOperator::PGOverlap and BinaryOperator::AND are both converted to Opeator::AND, this is what make `&&` and `AND` am alias.

But it seems to me that  BinaryOperator::PGOverlap (`&&`) to Operator::AND is more straightforward, maybe avoid  BinaryOperator::AND (`AND`) to Operator::AND is the one to fix?

Reply:
We have not introduced something like `a || b` as OR. If we need this, it will meet the similar issue.

Reply:
Postgres array overlap is the same as array intersect. Return common elements in two array

Reply:
Are you sure?

https://www.postgresql.org/docs/9.6/functions-array.html
> && | overlap (have elements in common) | ARRAY[1,4,3] && ARRAY[2,1] | t

```
postgres=# select array[1, 2] && array[3, 4];
 ?column? 
----------
 f
(1 row)

postgres=# select array[1, 2, 3] && array[3, 4];
 ?column? 
----------
 t
(1 row)
```

Reply:
Only the return value is different, both find if there is common between two.

However, whether we return array or boolean, we can still convert PGOverlap to Operator::Overlap for this operator.

Reply:
I prefer the behavior like Duckdb between we can still get the follow on information based on the return value, but we can't find out the common in arrays from boolean value.